src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -493,24 +462,27 @@ private void ResetLogAggregatorIfNeeded(Solution currentSolution)
                         {
                             var currentSolution = this.Processor.CurrentSolution;
 
-                            if (currentSolution != _processingSolution)
+                            if (currentSolution == _processingSolution)
                             {
-                                ResetLogAggregatorIfNeeded(currentSolution);
+                                return;
+                            }
 
-                                // clear version tracking set we already reported.
-                                _currentSnapshotVersionTrackingSet.Clear();
+                            // solution has changed
+                            ResetLogAggregatorIfNeeded(currentSolution);
 
-                                _processingSolution = currentSolution;
+                            _processingSolution = currentSolution;
 
-                                await RunAnalyzersAsync(this.Analyzers, currentSolution, (a, s, c) => a.NewSolutionSnapshotAsync(s, c), this.CancellationToken).ConfigureAwait(false);
+                            // synchronize new solution to OOP
+                            await currentSolution.Workspace.SynchronizePrimaryWorkspaceAsync(currentSolution, this.CancellationToken).ConfigureAwait(false);
```

Comment:
solution crawler runs only for primary solution. also method itself has check for it.